File: kitten-house-care/src/test/java/org/jesperancinha/housing/repository/OwnerRepositoryImplTest.java
package org.jesperancinha.housing.repository;

import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.jupiter.api.Test;
import reactor.core.publisher.Mono;

import java.io.IOException;

import static org.assertj.core.api.Assertions.assertThat;

class OwnerRepositoryImplTest {
    private final OwnerRepositoryImpl ownerRepository = new OwnerRepositoryImpl(new ObjectMapper());

    OwnerRepositoryImplTest() throws IOException {
    }

    @Test
    void givenLiable_checkLiability_Ok() {
        Mono<String> deVeluwe = ownerRepository.checkLiability("De Veluwe");
        final String block = deVeluwe.block();
        assertThat(block).isEqualTo("OK");
    }

    @Test
    void givenNonLiable_checkLiability_NOk() {
        Mono<String> the_swamp = ownerRepository.checkLiability("The Swamp");
        final String block = the_swamp.block();
        assertThat(block).isEqualTo("NOK");
    }
}

File: kitten-house-care/src/test/kotlin/org/jesperancinha/housing/repository/OwnerRepositoryImplTest.kt
package org.jesperancinha.housing.repository

import com.fasterxml.jackson.databind.ObjectMapper
import org.assertj.core.api.Assertions
import org.junit.jupiter.api.Test

internal class OwnerRepositoryImplTest {
    private val ownerRepository = OwnerRepositoryImpl(ObjectMapper())
    @Test
    fun givenLiable_checkLiability_Ok() {
        val deVeluwe = ownerRepository.checkLiability("De Veluwe")
        val block = deVeluwe.block()
        Assertions.assertThat(block).isEqualTo("OK")
    }

    @Test
    fun givenNonLiable_checkLiability_NOk() {
        val the_swamp = ownerRepository.checkLiability("The Swamp")
        val block = the_swamp.block()
        Assertions.assertThat(block).isEqualTo("NOK")
    }
}